controllers/upgrade_handlers.go

```diff
@@ -18,39 +18,125 @@ package controllers
 
 import (
 	"context"
+	"errors"
 	"fmt"
 
 	lcav1alpha1 "github.com/openshift-kni/lifecycle-agent/api/v1alpha1"
+
 	"github.com/openshift-kni/lifecycle-agent/controllers/utils"
+	"github.com/openshift-kni/lifecycle-agent/internal/backuprestore"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	ctrl "sigs.k8s.io/controller-runtime"
 )
 
+// handleUpgrade orchestrate main upgrade steps
 func (r *ImageBasedUpgradeReconciler) handleUpgrade(ctx context.Context, ibu *lcav1alpha1.ImageBasedUpgrade) (ctrl.Result, error) {
+	// log and update status to in-progress
```

Comment:
The overall flow is not that correct.  Comments for the high level, 
- The handler could be repeatedly called when waiting for long running process or if error happens. It shouldn't start from beginning everytime it's being called. 
- We shouldn't record every single error in the IBU status. Most of them could be just API call issues due to unstable network.

Reply:
Updated like we discussed offline